SVC-12072 Add Qualys EUS Detection Workaround

bsper2 opened the pull request.

bsper2 commented:
Tested on dt-login03

bsper2 commented:
One moment, there is one feature I want to add. (For some reason can't see where to mark this as a work-in-progress)

billglick commented on manifests/qualys_eus_reporting.pp:
```diff
@@ -0,0 +1,61 @@
+
+    exec { 'add_qualys_EUS_alias':
+      path    => '/bin:/usr/bin',
+      command => "sed -i \'\$a${bash_alias}\' /root/.bashrc",
```
What is the ‘\$a’ n this?

bsper2 replied:
`$a` makes it append it to the end of the .bashrc

billglick reviewed (commented):
Generally looks good. A couple comments/questions.

bsper2 commented on data/os/RedHat/8.4.yaml:
```diff
@@ -0,0 +1,12 @@
+---
```
Good call, added one for 8.2

billglick commented on data/os/RedHat/8.4.yaml:
```diff
@@ -0,0 +1,12 @@
+---
+
```
Would there be any logic in setting these outdated RHEL `8.2` & `8.4` versions to default to enabling `qualys_eus_reporting`?

```suggestion

profile_audit::qualys_eus_reporting::enabled: true

```

**Probably not**, because `CentOS`, `Rocky`, etc. could also be referencing these `os/RedHat/8.x.yaml` files.

bsper2 replied:
I was leaning towards no. First for the reason you mention about Centos/Rocky, but also since the 8.2, 8.4, and other 8.x yaml files will always be set in the module. They just act as placeholders to set the strings for whatever EUS repos we might need.

billglick commented on files/qualys_reporting:
```diff
@@ -0,0 +1,4 @@
+Defaults:qualys !mail_always
+Defaults:qualys !requiretty
+%qualys ALL=(ALL) NOPASSWD: NOMAIL: ALL
+qualys ALL=(ALL) NOPASSWD: NOMAIL: ALL
```
I suggest we move this to hiera data so that it could easily be customized if needed by a specific project's control repo.

bsper2 replied:
I was putting that in a file since I wasn't really expecting that config to ever really change that much per cluster. But yeah there will probably be some reason eventually so I can put that in hiera. For unsetting it, I'll probably go with a boolean to enable/disable qualys sudo, since that reads a little better than setting this hiera data to empty.

bsper2 replied:
This has been done.

billglick commented on manifests/qualys_eus_reporting.pp:
```diff
@@ -0,0 +1,81 @@
+      }
+    }
+
+    pam_access::entry { 'Allow sudo for qualys':
+      user       => 'qualys',
+      origin     => 'LOCAL',
+      permission => '+',
+      position   => '-1',
+    }
```
I think the `sudo` functionality for Qualys should be moved to the `qualys.pp` file/class. It makes sense to me that Qualys should be able to sudo by default, regardless of whether we are enabling EUS. 

If a particular deployment does not want the default sudo access for Qualys, they can simply unset the hiera value for the sudoers file that qualys would use.

e.g. I could see admin/provisioning type nodes in a cluster (log collectors, Puppet, xCAT, etc) possibly running the latest RHEL 8.x, while actual compute node type nodes being locked to a specific EUS version. In that case Qualys would distinguish between the types of nodes and will need to sudo on them regardless of EUS being setup.

billglick replied:
But I think that Qualys will be set to try to use sudo based on the definition in the Qualys config for that subnet. For nodes on these subnets that are not flagged in Puppet as EUS this may generate a bunch of email traffic about Qualys trying to sudeo and not having permission to.

bsper2 replied:
Oh yes I see what you're saying. Qualys elevated scans would be set on the subnet level, but as-is puppet wasn't fine grained enough to allow sudo separate from enabling the EUS fake reporting.

bsper2 replied:
This has been done.

billglick commented on manifests/qualys_eus_reporting.pp:
```diff
@@ -0,0 +1,81 @@
+    content => epp( "${module_name}/qualys_eus_reporting.sh.epp"),
+  }
+
+  sudo::conf { 'qualys_scan':
+    ensure   => $sudo_ensure_parm,
+    priority => 10,
+    content  => file("${module_name}/qualys_reporting"),
+  }
```
I think the `sudo` functionality for Qualys should be moved to the `qualys.pp` file/class. It makes sense to me that Qualys should be able to sudo by default, regardless of whether we are enabling EUS. 

If a particular deployment does not want the default sudo access for Qualys, they can simply unset the hiera value for the sudoers file that qualys would use.

e.g. I could see admin/provisioning type nodes in a cluster (log collectors, Puppet, xCAT, etc) possibly running the latest RHEL 8.x, while actual compute node type nodes being locked to a specific EUS version. In that case Qualys would distinguish between the types of nodes and will need to sudo on them regardless of EUS being setup.

bsper2 replied:
This has been done.

billglick reviewed (commented):
A couple more comments...

bsper2 commented:
Closing this pull request. I squashed my intermediate commits and force pushed but for some reason github still shows there are 16 commits. My local copy only shows the one I squashed into, and a git pull shows I'm up-to-date.
